graph/pull.go

```diff
@@ -28,10 +18,24 @@ type ImagePullConfig struct {
 	OutStream   io.Writer
 }
 
+type Puller interface {
+	Pull(image, name, tag string, endpoint registry.APIEndpoint, imagePullConfig *ImagePullConfig, sf *streamformatter.StreamFormatter) (fallback bool, err error)
+}
+
+func (s *TagStore) NewPuller(endpoint registry.APIEndpoint, repoInfo *registry.RepositoryInfo) (Puller, error) {
```

Comment:
@dmcgowan we already enumerated what's needed and yes I agree we should break that up into an interface and make Pull a function that takes that interface instead. However, we agreed to have that done in the next PR to avoid further complications in the review.